src/mod/applications/mod_av/avformat.c

```diff
@@ -2109,15 +2109,30 @@ GCC_DIAG_ON(deprecated-declarations)
 	return NULL;
 }
 
-static int avio_interrupt_cb(void *ctx) 
-{ 
+static int avio_interrupt_cb(void *ctx)
+{
 	av_file_context_t* context = (av_file_context_t*)ctx;
-	
+	switch_time_t current_time_us = switch_time_now();
+
 	if (context == NULL) {
 		return 0;
-	} else {
-		return context->closed;
 	}
+
+	if (context->closed) {
+		return 1;  // Interrupted due to close
+	}
+
+	// Check timeout if configured
+	if (!zstr(context->rw_timeout)) {
+		switch_time_t timeout_us = atoi(context->rw_timeout);
+		switch_time_t elapsed_us = current_time_us - context->connect_start_time;
+
+		if (elapsed_us > timeout_us) {
+			return 1;  // Interrupted due to timeout
```

Comment:
What happen when somehow a connect timeout fail? Will this remove the media bug or prune it?